src/h3/mod.rs

```diff
@@ -1400,7 +1400,17 @@ impl Connection {
 
 /// Generates an HTTP/3 GREASE variable length integer.
 fn grease_value() -> u64 {
-    let n = std::cmp::min(super::rand::rand_u64(), 148_764_065_110_560_899);
+    let range = 148_764_065_110_560_899;
+    let chunk_size = u64::max_value() / range;
+    let end_of_last_chunk = chunk_size * range;
+
+    let mut r = super::rand::rand_u64();
+
+    while r >= end_of_last_chunk {
```

Comment:
In this case, this condition only met very very tiny case...
2^64 - 148764065110560899 * (2^64 / 148764065110560899) = 140.
It's good enough I think.